conformance/tests/httproute-header-matching.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/gateway-api/conformance/utils/http"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, HTTPRouteHeaderMatching)
+}
+
+var HTTPRouteHeaderMatching = suite.ConformanceTest{
+	ShortName:   "HTTPRouteHeaderMatching",
+	Description: "A single HTTPRoute with header matching for different backends",
+	Manifests:   []string{"tests/httproute-header-matching.yaml"},
+	Test: func(t *testing.T, suite *suite.ConformanceTestSuite) {
+		ns := "gateway-conformance-infra"
+		routeNN := types.NamespacedName{Name: "matching", Namespace: ns}
```

Comment:
Nit: Maybe give this route a unique name to avoid any weird race conditions with the other route of the same name defined in another conformance test case.

```suggestion
		routeNN := types.NamespacedName{Name: "header-matching", Namespace: ns}
```